dev/archery/archery/benchmark/runner.py

```diff
@@ -120,11 +120,14 @@ def is_json_result(cls, path_or_str):
 
     @staticmethod
     def from_json(path_or_str, **kwargs):
-        # breaks recursive imports
-        from ..utils.codec import BenchmarkRunnerCodec
-        path_or_str, json_load = (open(path_or_str), json.load) \
-            if os.path.isfile(path_or_str) else (path_or_str, json.loads)
-        return BenchmarkRunnerCodec.decode(json_load(path_or_str), **kwargs)
+        # .codec imported here to break recursive imports
+        from .codec import BenchmarkRunnerCodec
```

Comment:
Oh! This is my bad. Thanks so much for fixing it!

Reply:
I do not understand the "break recursive imports" comment. I do not see how this import breaks the function from being called unless it has side-effects which I am not aware of.

Reply:
In Python, recursive imports are only bothersome at the top level. Otherwise they work just fine.